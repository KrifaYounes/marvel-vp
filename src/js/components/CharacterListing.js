import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Grid, Row } from 'react-flexbox-grid';
import { loadCharacters } from '../actions/characters-actions';
import Character from './Character';

class CharacterListing extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        this.props.loadCharacters();
    }

    render() {
        const characters = this.props.characters.map( (character) => {
                return ( <Character character={character} /> );
            });

        return (
            <div>
                { !this.props.isLoading ?
                    <Grid>
                        <Row>
                            {characters}
                        </Row>
                    </Grid>

                    : 'Loading Marvel Heroes'

                }
            </div>
        );
    }}


function mapStateToProps(state) {
    const characters = state.charactersReducer.characters;
    const isLoading = state.charactersReducer.isLoading;

    return {
        characters,
        isLoading
    };
}

export default connect(mapStateToProps, { loadCharacters } )(CharacterListing);

