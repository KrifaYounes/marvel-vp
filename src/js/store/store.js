import { createStore, applyMiddleware } from 'redux'
import { syncHistoryWithStore } from 'react-router-redux';
import { createHashHistory } from 'history';
import reducers from '../reducers/reducers';
import thunkMiddleware from 'redux-thunk';

export const history = createHashHistory();

const store = createStore(reducers, applyMiddleware(thunkMiddleware));

syncHistoryWithStore(history, store);

export default store;
