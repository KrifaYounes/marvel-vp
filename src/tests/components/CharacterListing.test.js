import React from 'react';
import CharacterListing from '../../js/components/CharacterListing';
import { shallow } from 'enzyme';
import enzymeToJSON from 'enzyme-to-json';
import store from '../../js/store/store';

test('CharacterListing Component Snapshot', () => {
    const characters = [
        {
            urls : [],
            thumbnail : {
                path : 'http://marvel.com/image',
                extension: 'jpeg'
            },
            descriptoin: 'Marvel Heroe Description',
            name: 'Marvel Heroe Name'
        },
        {
            urls : [],
            thumbnail : {
                path : 'http://marvel.com/image',
                extension: 'jpeg'
            },
            descriptoin: 'Marvel Heroe Description 2',
            name: 'Marvel Heroe Name 2'
        }

    ];

    const wrapper = shallow(<CharacterListing characters={characters} isLoading={false} store={store}/>);
    //must match the snapshot

    expect(enzymeToJSON(wrapper)).toMatchSnapshot();
});


test('CharacterListing Component Snapshot without characters', () => {
    const characters = null;

    const wrapper = shallow(<CharacterListing characters={characters} isLoading={false} store={store}/>);
    //must match the snapshot

    expect(enzymeToJSON(wrapper)).toMatchSnapshot();
});


test('CharacterListing Component Snapshot Loading', () => {
    const characters = null;

    const wrapper = shallow(<CharacterListing characters={characters} isLoading={true} store={store}/>);
    //must match the snapshot

    expect(enzymeToJSON(wrapper)).toMatchSnapshot();
});
