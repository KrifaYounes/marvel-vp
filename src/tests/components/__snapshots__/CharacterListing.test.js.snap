// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterListing Component Snapshot 1`] = `
<CharacterListing
  characters={Array []}
  isLoading={false}
  loadCharacters={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`CharacterListing Component Snapshot Loading 1`] = `
<CharacterListing
  characters={Array []}
  isLoading={false}
  loadCharacters={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`CharacterListing Component Snapshot without characters 1`] = `
<CharacterListing
  characters={Array []}
  isLoading={false}
  loadCharacters={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
